CMakeLists.txt

```diff
@@ -151,20 +151,22 @@ if(ENABLE_BUILD_VMM)
         DEPENDS bfunwind
     )
 
-    add_subproject(
-        bfvmm_main vmm
+    vmm_extension(
+        bfvmm_main
         DEPENDS bfvmm
         DEPENDS bfintrinsics
-        SOURCE_DIR ${SOURCE_BFVMM_DIR}/src/main
-    )
-
-    add_subproject(
-        integration vmm
-        DEPENDS bfvmm
-        SOURCE_DIR ${CMAKE_CURRENT_LIST_DIR}/bfvmm/integration/
+        SOURCE_DIR ${SOURCE_EXAMPLE_DIR}/bfvmm_main/src
     )
 endif()
 
+# if(ENABLE_BUILD_TEST AND ENABLE_BUILD_VMM)
```

Comment:
I am fine with working on the integration tests one by one so long as the test_all.cpp test is maintained and works so that we can see an example of how its going to work, and the test_all.cpp is a good one to ensure that the APIs are working.